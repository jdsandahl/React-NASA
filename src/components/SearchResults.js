import React from "react";
import PropTypes from 'prop-types';
import '../styles/SearchResults.css';

const SearchResults = ({ image }) => {
  console.log(image);
  return (
    <div className="SearchResults" data-testid="SearchResults">
      <img
        className="image-card"
        src={ image }
        alt={ image }
        data-testid="image-card"
      />
    </div>
  );
};

SearchResults.propTypes = {
    image: PropTypes.string.isRequired,
}

export default SearchResults;
