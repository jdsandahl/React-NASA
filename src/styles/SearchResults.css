.SearchResults {
    display: grid;
    grid-column: 1fr 1fr 1fr 1fr;
    grid-row: auto;
    margin: 25px;
}

.image-card {
    border-radius: 35px;
    max-width: 300px;
    max-height: 300px;
    padding: 25px;  
}