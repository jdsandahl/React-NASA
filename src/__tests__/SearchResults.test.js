import React from 'react';
import { render } from '@testing-library/react';
import SearchResults from '../components/SearchResults.js';

describe('SearchResults', () => {
    
    test('renders correctly to match snapshot', () => {
        const { asFragment } = render(<SearchResults image="mockImage" />);    
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the correct elements', () => {
        const { queryByTestId, queryByAltText } = render(<SearchResults image="mockImage" />);

        expect(queryByTestId('SearchResults')).toBeTruthy();
        expect(queryByTestId('image-card')).toBeTruthy();
        expect(queryByAltText("mockImage")).toBeTruthy();

    });
});