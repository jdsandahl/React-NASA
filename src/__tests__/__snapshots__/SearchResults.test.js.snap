// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchResults"
    data-testid="SearchResults"
  >
    <img
      alt="mockImage"
      class="image-card"
      data-testid="image-card"
      src="mockImage"
    />
  </div>
</DocumentFragment>
`;
